src/utils/utils.ts

```diff
@@ -313,3 +313,16 @@ export function getUserAgent() {
   const packageJson = require('./../../package.json');
   return `Playwright/${packageJson.version} (${os.arch()}/${os.platform()}/${os.release()})`;
 }
+
+export function constructURLBasedOnBaseURL(baseURL: string | undefined, givenURL: string): string {
```

Comment:
Hmm...  I'd make it safer this way:

```js
try {
  return (new URL.URL(givenURL, baseURL)).toString();
} catch (e) {
  return givenURL;
}
```

Why do we need to special-case '/' or ''? Same for `!baseURL`.